polysh/host_syntax.py

```diff
@@ -64,4 +70,4 @@ def expand_syntax(string):
                     for expanded in expand_syntax(prefix + i + suffix):
                         yield expanded
     else:
-        yield string
+        yield _split_port(string)
```

Comment:
This changes the return type of expand_syntax() from str to tuple. There is more then one place calling this method though. Once in the if block above, which I don't really know what it's doing. Looks like it might be fine though. Also it's used in selected_shells() and it doesn't look like that place knows how to handle the new return type. This could be why the tests don't pass right now.

Reply:
Sadly it's one of those cases where polysh crashes way down the road, making it hard to see what's exactly going wrong here. I'm pretty sure it's `if fnmatch(i.display_name, expanded_pattern):` in selected_shells() not working as expected anymore though. It will never match as `expanded_pattern` is that tuple now.
```
╭─space@FeTAp ~/repos/ig/polysh  ‹master›
╰─$ ./run.py localhost                                                                          1 ↵
ready (1)> :reset_prompt localhost
error: uncaptured python exception, closing channel <polysh.stdin.SocketNotificationReader connected '' at 0x10dbfdf98> (<class 'TypeError'>:expected str, bytes or os.PathLike object, not tuple [/usr/local/Cellar/python/3.6.4_4/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncore.py|readwrite|108] [/usr/local/Cellar/python/3.6.4_4/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncore.py|handle_read_event|423] [/Users/space/repos/ig/polysh/polysh/stdin.py|handle_read|139] [/Users/space/repos/ig/polysh/polysh/stdin.py|_do|124] [/Users/space/repos/ig/polysh/polysh/stdin.py|process_input_buffer|69] [/Users/space/repos/ig/polysh/polysh/control_commands_helpers.py|handle_control_command|121] [/Users/space/repos/ig/polysh/polysh/control_commands.py|do_reset_prompt|93] [/Users/space/repos/ig/polysh/polysh/control_commands_helpers.py|selected_shells|59] [/usr/local/Cellar/python/3.6.4_4/Frameworks/Python.framework/Versions/3.6/lib/python3.6/fnmatch.py|fnmatch|35] [/usr/local/Cellar/python/3.6.4_4/Frameworks/Python.framework/Versions/3.6/lib/python3.6/posixpath.py|normcase|52])
Traceback (most recent call last):
  File "./run.py", line 22, in <module>
    run()
  File "/Users/space/repos/ig/polysh/polysh/main.py", line 258, in run
    loop(args.interactive)
  File "/Users/space/repos/ig/polysh/polysh/main.py", line 179, in loop
    stdin.the_stdin_thread.want_raw_input()
  File "/Users/space/repos/ig/polysh/polysh/stdin.py", line 258, in want_raw_input
    self.socket_notification.handle_read()
  File "/Users/space/repos/ig/polysh/polysh/stdin.py", line 139, in handle_read
    self._do(c)
  File "/Users/space/repos/ig/polysh/polysh/stdin.py", line 126, in _do
    raise Exception('Unknown code: %s' % (c))
Exception: Unknown code: b''
```

Reply:
thanks for the hint, I will look after it.